CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/CollectionRequestHandler.swift

```diff
@@ -0,0 +1,142 @@
+            let collection: Collection = try database.createCollection(name: collectionName, scope: scopeName)
+            return collection
+            
+        case "collection_collectionNames":
+            let scope: String = (args.get(name:"scopeName"))!
+            let database: Database = (args.get(name:"database"))!
+            var names = [String]()
+            let collectionNames = try database.collections(scope: scope)
+            for collection in collectionNames{
+                let collectionObject: Collection = collection
+                names.append(collectionObject.name)
+            }
+            return names
+        
+        case "collection_deleteCollection":
+            let scopeName: String = (args.get(name:"scopeName"))!
+            let database: Database = (args.get(name:"database"))!
+            let collectionName: String = (args.get(name:"collectionName"))!
+            do{
+                try database.deleteCollection(name: collectionName, scope: scopeName)
+            } catch{
+                return error
+            }
+        
+        case "collection_saveDocument":
+            let document: MutableDocument = (args.get(name:"document"))!
+            let collection: Collection = (args.get(name:"collection"))!
+            do {
+                try collection.save(document: document)
+            } catch {
+                return error
+            }
+        
+        case "collection_getDocument":
+            let docId: String = (args.get(name: "docId"))!
+            let collection : Collection = (args.get(name: "collection"))!
+            return try collection.document(id: docId)
+        
+        case "collection_deleteDocument":
+            let document: Document = (args.get(name: "document"))!
+            let collection: Collection = (args.get(name: "collection"))!
+            do {
+                try collection.delete(document: document)
+            } catch {
+                return error
+            }
+        
+        case "collection_purgeDocument":
+            let document: Document = (args.get(name: "document"))!
+            let collection: Collection = (args.get(name: "collection"))!
+            do {
+                try collection.purge(document: document)
+            } catch {
+                return error
+            }
+        
+        case "collection_purgeDocumentID":
+            let docID: String = (args.get(name: "docId"))!
+            let collection: Collection = (args.get(name: "collection"))!
+            do {
+                try collection.purge(id: docID)
+            } catch {
+                return error
+            }
+        
+        case "collection_getDocumentExpiration":
+            let docId: String = (args.get(name: "docId"))!
+            let collection: Collection = (args.get(name: "collection"))!
+            do {
+                let date = try collection.getDocumentExpiration(id: docId)
+                return date
```

Comment:
you could drop the try/catch, since its already handled by the throwing function? also result is returned